apps/web/src/actions/user-profile-mod.ts

```diff
@@ -3,48 +3,186 @@
 import { authenticatedAction } from "@/lib/safe-action";
 import { z } from "zod";
 import { db } from "db";
-import { userCommonData } from "db/schema";
+import { userCommonData, userHackerData } from "db/schema";
 import { eq } from "db/drizzle";
 import { put } from "@vercel/blob";
 import { decodeBase64AsFile } from "@/lib/utils/shared/files";
 import { revalidatePath } from "next/cache";
-import { getUser } from "db/functions";
+import { getUser, getUserByTag } from "db/functions";
+import { RegistrationSettingsFormValidator } from "@/validators/shared/RegistrationSettingsForm";
 
-// TODO: Add skill updating
 export const modifyRegistrationData = authenticatedAction(
+	RegistrationSettingsFormValidator,
+	async (
+		{
+			age,
+			gender,
+			race,
+			ethnicity,
+			isEmailable,
+			university,
+			major,
+			levelOfStudy,
+			schoolID,
+			hackathonsAttended,
+			softwareBuildingExperience,
+			heardAboutEvent,
+			shirtSize,
+			dietaryRestrictions,
+			accommodationNote,
+			github,
+			linkedin,
+			personalWebsite,
+			phoneNumber,
+			countryOfResidence,
+		},
+		{ userId },
+	) => {
+		const user = await db.query.userCommonData.findFirst({
+			where: eq(userCommonData.clerkID, userId),
+		});
+		if (!user) throw new Error("User not found");
+		await db
+			.update(userCommonData)
+			.set({
+				age,
+				gender,
+				race,
+				ethnicity,
+				shirtSize,
+				dietRestrictions: dietaryRestrictions,
+				accommodationNote,
+				phoneNumber,
+				countryOfResidence,
+			})
+			.where(eq(userCommonData.clerkID, user.clerkID));
+		await db
+			.update(userHackerData)
+			.set({
+				isEmailable,
+				university,
+				major,
+				levelOfStudy,
+				schoolID,
+				hackathonsAttended,
+				softwareExperience: softwareBuildingExperience,
+				heardFrom: heardAboutEvent,
+				GitHub: github,
+				LinkedIn: linkedin,
+				PersonalWebsite: personalWebsite,
+			})
+			.where(eq(userHackerData.clerkID, user.clerkID));
+		return {
+			success: true,
+			newAge: age,
+			newGender: gender,
+			newRace: race,
+			newEthnicity: ethnicity,
+			newWantsToReceiveMLHEmails: isEmailable,
+			newUniversity: university,
+			newMajor: major,
+			newLevelOfStudy: levelOfStudy,
+			newSchoolID: schoolID,
+			newHackathonsAttended: hackathonsAttended,
+			newSoftwareExperience: softwareBuildingExperience,
+			newHeardFrom: heardAboutEvent,
+			newShirtSize: shirtSize,
+			newDietaryRestrictions: dietaryRestrictions,
+			newAccommodationNote: accommodationNote,
+			newGitHub: github,
+			newLinkedIn: linkedin,
+			newPersonalWebsite: personalWebsite,
+			newPhoneNumber: phoneNumber,
+			newCountryOfResidence: countryOfResidence,
+		};
+	},
+);
+
+export const modifyResume = authenticatedAction(
 	z.object({
```

Comment:
Same with this one here in terms of `Promise.all()` if possible with the other stuff.